services/QuillLMS/app/models/subscription.rb

```diff
@@ -80,15 +83,26 @@ class Subscription < ApplicationRecord
   SCHOOL_FIRST_PURCHASE_PRICE = SCHOOL_RENEWAL_PRICE
   TEACHER_PRICE = 8000
   ALL_PRICES = [TEACHER_PRICE, SCHOOL_RENEWAL_PRICE]
-  PAYMENT_METHODS = ['Invoice', 'Credit Card', 'Premium Credit']
+  PAYMENT_METHODS = [
+    INVOICE_PAYMENT_METHOD = 'Invoice',
+    CREDIT_CARD_PAYMENT_METHOD = 'Credit Card',
+    PREMIUM_CREDIT_PAYMENT_METHOD = 'Premium Credit'
+  ]
+
   ALL_TYPES = OFFICIAL_FREE_TYPES.dup.concat(OFFICIAL_PAID_TYPES)
 
+  validates :stripe_invoice_id, allow_blank: true, stripe_uid: { prefix: :in }
+
   scope :active, -> { where(de_activated_date: nil).where("expiration > ?", Date.today).order(expiration: :asc) }
 
   def is_trial?
     account_type && TRIAL_TYPES.include?(account_type)
   end
 
+  def expired?
```

Comment:
I think we're using `expiration <= Date.today` in some other places in the code to check if a subscription is expired. Might be worth double checking to make sure we're using this new method in all those old places.

Reply:
That's a good point.  Dan made a [suggestion](https://github.com/empirical-org/Empirical-Core/pull/9005#discussion_r845563873) to change this so I did but I will probably revert to the original and then change all of them when adding in the Rubocop configuration.